android/src/main/java/io/branch/rnbranch/RNBranchModule.java

```diff
@@ -60,22 +65,55 @@ public void onInitFinished(JSONObject referringParams, BranchError error) {
                 Log.d(REACT_CLASS, "onInitFinished");
                 JSONObject result = new JSONObject();
                 try{
-                    result.put("params", referringParams != null && referringParams.has("~id") ? referringParams : JSONObject.NULL);
-                    result.put("error", error != null ? error.getMessage() : JSONObject.NULL);
-                    result.put("uri", uri != null ? uri.toString() : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_PARAMS, referringParams != null && referringParams.has("~id") ? referringParams : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_ERROR, error != null ? error.getMessage() : JSONObject.NULL);
+                    result.put(NATIVE_INIT_SESSION_FINISHED_EVENT_URI, uri != null ? uri.toString() : JSONObject.NULL);
                 } catch(JSONException ex) {
                     try {
                         result.put("error", "Failed to convert result to JSONObject: " + ex.getMessage());
                     } catch(JSONException k) {}
                 }
                 initSessionResult = result;
-                LocalBroadcastManager.getInstance(mmActivity).sendBroadcast(new Intent(NATIVE_INIT_SESSION_FINISHED_EVENT));
+
+                generateLocalBroadcast(referringParams, uri, error);
             }
 
             private Branch.BranchReferralInitListener init(Activity activity) {
                 mmActivity = activity;
                 return this;
             }
+
+            private void generateLocalBroadcast(JSONObject referringParams, Uri uri, BranchError error) {
+                Intent broadcastIntent = new Intent(NATIVE_INIT_SESSION_FINISHED_EVENT);
+
+                if (referringParams != null) {
+                    broadcastIntent.putExtra(NATIVE_INIT_SESSION_FINISHED_EVENT_PARAMS, referringParams.toString());
+
+                    if (referringParams.has("~id")) {
+                        BranchUniversalObject branchUniversalObject = BranchUniversalObject.createInstance(referringParams);
+                        // BranchUniversalObject branchUniversalObject = BranchUniversalObject.getReferredBranchUniversalObject();
+                        if (branchUniversalObject != null) {
+                            broadcastIntent.putExtra(NATIVE_INIT_SESSION_FINISHED_EVENT_BRANCH_UNIVERSAL_OBJECT, branchUniversalObject);
+                        }
+
+                        // TODO: Can this be constructed with local state?
+                        LinkProperties linkProperties = LinkProperties.getReferredLinkProperties();
```

Comment:
I suppose I usually prefer to use the callback context rather than stored state when handling asynchronous events. For example, there can be races. But a race is not much of a practical concern here, since these are user-initiated events. At any rate, I'd like things to be consistent. I can make them both use the latest_referring_params.